codespan/src/codemap.rs

```diff
@@ -5,7 +5,7 @@ use std::sync::Arc;
 use filemap::{FileMap, FileName};
 use index::{ByteIndex, ByteOffset};
 
-#[derive(Debug, Default)]
+#[derive(Clone, Debug, Default)]
 pub struct CodeMap {
     files: Vec<Arc<FileMap>>,
```

Comment:
I wonder if the `Vec` could also be `Arc`ed? Might have to revisit what Rust does - I think I simplified things here.

Reply:
It could and then one would use `make_mut` to get a unique reference to mutate it. But that could be done as `Arc<CodeMap>` as well so.